Added proper right-to-left language support.

DannyVancura opened the pull request:
Implemented different behavior for the card text field when running in Right-to-left language environment.

* Modified view animation to crop the card number to the last four digits when RTL.
* Inverted reading of card type image, card number, month, year, CVC and accessory button when RTL.
* To test this when running on simulator in a left-to-right language, you can do the following:

1. Clone the `Caishen_Example` scheme. 
2. Edit the cloned scheme: `Run ‣ Options ‣ Application Language` - change that to `Right to Left Pseudolanguage`.
3. Run that scheme.

@mohammad19991

mohammad19991 commented:
@DannyVancura is RTL support ready?

DannyVancura commented:
@mohammad19991 the changes should reflect what I showed in the mockup image. If you could checkout the PR and let me know whether the look & feel is right, that would help a lot!

mohammad19991 commented:
@DannyVancura hey! it's working as the mockup which is good thanks, but there are 2 issues:
1. When swipe from CVV, data to card number: kind of overlapping happens between the numbers
2. the textfield outlet crash in IB, check attached image which shows where the crash happens.

<img width="826" alt="screen shot 2016-04-28 at 09 43 52" src="https://cloud.githubusercontent.com/assets/2028141/14876460/2a49bd04-0d26-11e6-9756-cbb041c8b43e.png">

DannyVancura commented:
@mohammad19991 Thank you for the review! I fixed the crash and changed the view animation to show the full card number after the view animation has ended to avoid the overlapping.
Can you check if there are any other issues to be addressed on this PR?

mohammad19991 commented:
Hi @DannyVancura it's better now, but the overlapping issue still happens so to reproduce this issue **follow the below steps:**
1. fill all the information.
2. swipe left to card number.
3. again swipe right to CVV and Expiry date.
4. swipe right again and observe the result.

**Check attached image to see this problem:**
![img_2262](https://cloud.githubusercontent.com/assets/2028141/14940515/30673124-0f8a-11e6-9501-ee8bc60b2599.PNG)

one more thing, if you want to make this library fully localized you have to make sure others screens are also supports RTL like this screen:

![simulator screen shot may 1 2016 10 38 21](https://cloud.githubusercontent.com/assets/2028141/14940522/69774d1e-0f8a-11e6-8e8e-78a4e60d7bfa.png)

also if you need help with the translations let me know.

DannyVancura commented:
Hey @mohammad19991 ! Thank you for the very detailed response!
The bugs you mentioned should be fixed now.

As for the other view you mentioned (second picture): This is part of [Card.io](https://github.com/card-io/card.io-iOS-SDK) and not included in the Caishen itself but as an example implementation.

If you can help out with any language support, that help would be highly appreciated! Feel free to create a pull request with whatever language you can help out with! You can find examples for the localization of VoiceOver accessibility attributes [here](https://github.com/prolificinteractive/Caishen/tree/master/Pod/Resources).

mohammad19991 commented:
@DannyVancura yea the bugs are fixed thank you, and regarding the language support I will be happy to help.

JamesJSY commented on Pod/Classes/UI/CardTextField.swift:
```diff
@@ -244,13 +267,16 @@ public class CardTextField: UITextField, NumberInputTextFieldDelegate {
+        
         let leftSwipeGestureRecognizer = UISwipeGestureRecognizer(target: self, action: #selector(moveNumberFieldLeftAnimated))
-        leftSwipeGestureRecognizer.direction = .Left
+        leftSwipeGestureRecognizer.direction = isRightToLeftLanguage ? .Right : .Left
```
Just an idea. Would something like `swipeToDetailInputTextFieldGestureRecognizer` be a better name? Same as for the `rightSwipeGestureRecognizer` vs `swipeToNumberInputTextFieldGestureRecognizer`

DannyVancura replied:
Makes sense, I'll change that. Thanks!

ThibaultKlein commented on Pod/Classes/UI/CardTextField.swift:
```diff
@@ -283,6 +309,18 @@ public class CardTextField: UITextField, NumberInputTextFieldDelegate {
+            cvcTextField.textAlignment = .Right
+            monthTextField.textAlignment = .Right
+            yearTextField.textAlignment = .Left
+            numberInputTextField.textAlignment = .Left
```
can't you use the `Natural` option?

ThibaultKlein replied:
seems that you changed the xib to use `textAlignment="natural"`, why do you need this snippet then? Is it to support version prior to iOS 9?

DannyVancura replied:
Unfortunately the `Natural` option does not have an equivalent for `Right`. I could have used Natural for `yearTextField` and `numberInputTextField` but thought this implementation is more descriptive of what is happening.

Ideally, I would have gone with `Natural` and an `NaturalInverted` - only that the latter does not exist.

ThibaultKlein replied:
cf Apple documentation: 

> Use the default alignment associated with the current localization of the app. The default alignment for left-to-right scripts is NSTextAlignmentLeft, and the default alignment for right-to-left scripts is NSTextAlignmentRight.

So it should work if your app is using a language that requires a RTL alignment right?

ThibaultKlein commented:
👍 but it would be good to research a little bit more about the `Natural` option to simplify the code.

DannyVancura commented:
Thanks, I updated that and commented the respective code to clarify the use of manually set text alignment.
